### ISSUE TITLE
Improve external media experience in media library

### ISSUE THREAD

**Issue by johngodley** [Automattic/wp-calypso#31530]
The media library provides a way to navigate media from external services such as Google Photos and Pexels:

<img width="258" alt="media_library" src="https://user-images.githubusercontent.com/1277682/54533480-2feabd80-4982-11e9-9ea2-1255af6d06bd.png">

External media needs to be copied to the user's media library before inserting, and this is a two-step process:

1. Select media in external library and click 'copy' to copy to WordPress library
2. Insert media from WordPress library into post

It would be better if this was as seamless as inserting WordPress media. That is, you select the media and it copies and pastes in one action.

This used to work with the classic editor, but the way that Gutenberg interacts with the media library means it's no longer possible.

As part of future changes to the media library it would be good to try and improve the Calyspo/Gutenberg external media library situation.

**Comment by JoshuaGoode** [comment]
Encountered issue related to this on 1881591-zen

1. Simple Gutenberg site.
2. Edit page/post. Insert image via Google Photos library.
3. Instead of copying over, Image links directly to Google's servers and eventually 403s for visitors and breaks the image.

Provided temporary workaround of accessing My Site > Media and copying the images over there first before inserting into page/post.

Another instance of this was also reported on 1887293-zen

**Comment by JoshuaGoode** [comment]
Since all new WPCOM users defaulted to Gutenberg, any instance of a user trying to use the Google Photo library directly in a post will have their image links broken. Should this be marked as a bug with increased priority? @johngodley

**Comment by johngodley** [comment]
This issue is to improve the experience of external media in Calypso and Gutenberg.

The bug you mention is described in #30953, and will be dealt with there. It has been in effect since Gutenberg was introduced to WPCOM, so I don't think changing priority will make a lot of difference.

**Comment by hubertnguyen** [comment]
Just wanted to chime in and say that in my opinion, adding a media via URL without copying it to the local file system is a good thing.

The user already has the option of uploading a file, like we always did in Wordpress. However, externally hosted images have second-class support today. It's great that Gutenberg allows the inclusion of externally hosted files into a post. 

However, there's no support for galleries made of externally hosted images. 

Yes, I understand that we might not have thumbnails and stuff like that, but we also have CDNs that can resize images on the fly. 

Using an external source can be great for sharing assets between different WordPress installs of one company, or it may allow someone to license hosted images, and control the rights from one place. It also makes Wordpress backups, restores, and recovery MUCH faster. 

In short, there are many great benefits of having images stored outside of the local WP install with full WP Media Library support. Please consider this. 

Thank you!

**Comment by johngodley** [comment]
You can already reference external images using the 'insert from URL' option.

Specifically for the Google Photos image library this is not an option as it is not possible to publicly link to media in your library.

**Comment by johngodley** [comment]
Ok, then this is probably not the best place to make that request. This is a bug report for copying media from Google Photos or Pexels.

I'm not sure if you are talking about WordPress.com or WordPress.org, but you can make a feature request for WordPress.com [here](https://github.com/Automattic/wp-calypso/issues), or one for the WordPress.org Gutenberg editor [here](https://github.com/WordPress/gutenberg/).

**Comment by hubertnguyen** [comment]
Got it. I thought that WP Media was outside of Gutenberg, but maybe I'll try WP.com Thank you.
